pkg/telemetry/telemetry.go

```diff
@@ -327,6 +347,59 @@ func (t *engineImpl) getInstanceInformationStats() (Metadata, error) {
 	return stats, nil
 }
 
+// Let's keep this high as large systems can take a good time evaluating histograms.
+const promqlQueryTimeout = time.Minute * 10
+
+func (t *engineImpl) syncPromqlTelemetry() {
+	if t.promqlQueryable == nil {
+		// When not testing PromQL stats, let's skip this function.
+		return
+	}
+	stats := make(Metadata)
+	for i := range promqlStats {
+		if promqlStats[i].parsedQuery == nil {
+			qry, err := t.promqlEngine.NewInstantQuery(t.promqlQueryable, promqlStats[i].query, time.Now())
+			if err != nil {
+				log.Debug("msg", "error creating instant promql query", "err", err.Error())
+				continue
+			}
+			promqlStats[i].parsedQuery = qry
+		}
+
+		result := promqlStats[i].parsedQuery.Exec(context.Background())
+		if result.Err != nil {
+			log.Debug("msg", "error executing promql query", "query", promqlStats[i].query, "err", result.Err.Error())
+			continue
+		}
+
+		value, err := getValue(result)
+		if err != nil {
+			log.Debug("msg", "error extracting value from response", "query", promqlStats[i].query, "err", err.Error())
+			continue
+		}
```

Comment:
All this logic (along with the getValue function) is better as a method on `promqlTelemetry`. The engine should be able to call promqlStats[I].Execute() and get the value back? Why -- because the engine should know nothing about internal structures like parsedQuery